### PR TITLE
Fixes #370: Deprecation warnings on Django 3.x

### LINKED ISSUE #370: Deprecation warnings with Django 3.1
In my unittests, I discovered some deprecation warnings:

.../python3.8/site-packages/mozilla_django_oidc/urls.py:21: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
  url(r'^callback/$', OIDCCallbackClass.as_view(),

This should be easy to fix for all Django versions 2.0 and greater (replace by path syntax). I can provide a PR, do you still support Django 1.11 and Python 2.7 or can this just be replaced?

.../python3.8/site-packages/mozilla_django_oidc/views.py:130: RemovedInDjango40Warning: django.utils.http.is_safe_url() is deprecated in favor of url_has_allowed_host_and_scheme().
  is_safe = is_safe_url(**kwargs)

This is only a rename and works for all version of Django >= 3.0 - maybe, this can be fixed later, when Django 2.2 is EOL.

### PR DESCRIPTION
**Pull request by jannh**
Since Django 1.11, which is the last version not supporting the new path syntax is not supported anymore, there should be no blockers merging this.
Btw: I found some import try/except blocks for python2/3 - if python2 is unsupported, should those be removed?

### CODE REVIEW

**Comment by akatsoulas** [comment]
> Btw: I found some import try/except blocks for python2/3 - if python2 is unsupported, should those be removed?
Sounds good to me :+1: 

@jannh Tests are failing b/c there are references of Django 1.11 that should be removed

**Comment by jannh** [comment]
I now fixed the test config (only Django >= 2.2 and corresponding supported python versions), fixed deprecation warnings and removed the python2 import shims.

**Review by akatsoulas — COMMENTED** [review]
Thanks @jannh for this PR! I did a first pass and this looks good to me. I will have another more thorough one before merging. I won't merge it yet b/c this has backwards incompatible changes. I will do another release of the lib first and then bump the major version with this PR.

**Comment by jannh** [comment]
Any news on this PR - it's been around for a while now :)

File: .travis.yml [hunk 1]
```diff
@@ -5,8 +5,6 @@ sudo: false
 language: python
 
 python:
-  - "2.7"
-  - "3.4"
   - "3.5"
```
Comment (sbor23) [inline .travis.yml:8]: 3.5 is EOL by now.

File: tox.ini [hunk 2]
```diff
@@ -1,20 +1,16 @@
 [tox]
 envlist =
     lint
-    py{27,34,35,36}-django111
-    py{34,35,36,37}-django200
-    py{35,36,37}-django210
-    py{35,36,37}-django220
+    py{35,36,37,38}-django220
```
Comment (sbor23) [inline tox.ini:4]: drop 35, add 39

File: tox.ini [hunk 3]
```diff
@@ -1,20 +1,16 @@
...
     py{36,37,38}-django300
+    py{36,37,38}-django310
```
Comment (sbor23) [inline tox.ini:6]: add 39

File: tox.ini [hunk 4]
```diff
@@ -1,20 +1,16 @@
...
 [travis]
 python =
-  2.7: py27
-  3.4: py34
   3.5: py35
   3.6: py36
-  3.7: py37, coverage, lint
-  3.8: py38
+  3.7: py37
+  3.8: py38, coverage, lint
```
Comment (sbor23) [inline tox.ini:13]: same here

**Review by sbor23 — COMMENTED** [review]
Hey, some remarks since this PR was lying around for long...
Needs a rebase on master too, there are some conflicts.

I really hope this finally gets some attention by the maintainers.

File: .travis.yml (hunk 1 above)
Comment (jannh) [reply .travis.yml]: Updated PR: Removed 3.5 and added 3.9

File: tox.ini (hunk 2 above)
Comment (jannh) [reply tox.ini]: Updated PR: Removed 3.5 and added 3.9

File: tox.ini (hunk 3 above)
Comment (jannh) [reply tox.ini]: Updated PR: Removed 3.5 and added 3.9

File: tox.ini (hunk 4 above)
Comment (jannh) [reply tox.ini]: Updated PR: Removed 3.5 and added 3.9

**Comment by jannh** [comment]
Removed python 3.5, added python 3.9 to tests.
Rebased to current master branch

File: tox.ini [hunk 5]
```diff
@@ -1,20 +1,16 @@
...
+    py{36,37,38,39}-django300
+    py{36,37,38,39}-django310
 
 [travis]
 python =
-  2.7: py27
-  3.4: py34
-  3.5: py35
   3.6: py36
-  3.7: py37, coverage, lint
```
Comment (johngian) [inline tox.ini:16]: This should be adapted in the codecov gh action. But i don't think it should block this PR.

**Review by akatsoulas — APPROVED** [review]
Thanks @jannh for this PR. :rocket:

**Comment by akatsoulas** [comment]
Closing in favor of #403 which removes travis

**Pull request closed without merging**